site24x7/user_data_source.go

```diff
@@ -0,0 +1,101 @@
+package site24x7
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/site24x7/terraform-provider-site24x7/api"
+)
+
+var userDataSourceSchema = map[string]*schema.Schema{
+	"name_regex": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Regular expression denoting the name of the user.",
+	},
+	"matching_ids": {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Description: "List of user IDs matching the name_regex.",
+	},
+	"matching_ids_and_names": {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Description: "List of user IDs and names matching the name_regex.",
+	},
+	"display_name": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Display name for the user.",
+	},
+	"email": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Email address of the user.",
+	},
+	"role": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Role assigned to the user.",
+	},
+	"status": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Status of the user (active or inactive).",
+	},
+}
+
+func DataSourceSite24x7User() *schema.Resource {
+	return &schema.Resource{
+		Read:   userDataSourceRead,
+		Schema: userDataSourceSchema,
+	}
+}
+
+// userDataSourceRead fetches users from Site24x7 based on name_regex
+func userDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(Client)
+
+	userList, err := client.Users().List()
+	if err != nil {
+		return err
+	}
+
+	nameRegex := d.Get("name_regex").(string)
+	var matchingUserIDs []string
+	var matchingUserIDsAndNames []string
```

Comment:
Use map or hashmap or tuple datastructure

Reply:
[Refactor matchingUserIDsAndNames to use map data structure](https://github.com/site24x7/terraform-provider-site24x7/pull/317/commits/53b4fac91fa5921f1881a297de724b3b0b802b8c)